newIDE/app/src/AssetStore/AssetStoreNavigator.js

```diff
@@ -0,0 +1,151 @@
+// @flow
+import * as React from 'react';
+import { type FiltersState } from '../UI/Search/FiltersChooser';
+import {
+  type AssetShortHeader,
+  type AssetPack,
+} from '../Utils/GDevelopServices/Asset';
+
+export type AssetStorePageState = {|
+  isOnHomePage: boolean,
+  openedAssetPack: ?AssetPack,
+  openedAssetShortHeader: ?AssetShortHeader,
+  filtersState: FiltersState,
+  ignoreTextualSearch: boolean,
+|};
+
+export type NavigationState = {|
+  previousPages: Array<AssetStorePageState>,
+  getCurrentPage: () => AssetStorePageState,
+  backToPreviousPage: () => void,
+  openHome: () => void,
+  clearHistory: () => void,
+  openSearchIfNeeded: () => void,
+  openTagPage: string => void,
+  openPackPage: AssetPack => void,
+  openDetailPage: AssetShortHeader => void,
+|};
+
+export const assetStoreHomePageState: AssetStorePageState = {
+  isOnHomePage: true,
+  openedAssetShortHeader: null,
+  openedAssetPack: null,
+  filtersState: {
+    chosenCategory: null,
+    chosenFilters: new Set(),
+    addFilter: () => {},
+    removeFilter: () => {},
+    setChosenCategory: () => {},
+  },
+  ignoreTextualSearch: false,
+};
+
+type AssetStorePageHistory = {|
+  previousPages: Array<AssetStorePageState>,
+|};
+
+export const useNavigation = (): NavigationState => {
+  const [history, setHistory] = React.useState<AssetStorePageHistory>({
+    previousPages: [assetStoreHomePageState],
+  });
+  const previousPages = history.previousPages;
+
+  return {
+    previousPages,
+    getCurrentPage: () => previousPages[previousPages.length - 1],
+    backToPreviousPage: () => {
+      if (previousPages.length > 1) {
+        previousPages.pop();
+        setHistory({ previousPages });
+      }
+    },
+    openHome: () => {
+      previousPages.length = 0;
+      previousPages.push(assetStoreHomePageState);
+      setHistory({ previousPages });
+    },
+    clearHistory: () => {
+      const currentPage = previousPages[previousPages.length - 1];
+      previousPages.length = 0;
+      previousPages.push(assetStoreHomePageState);
+      previousPages.push(currentPage);
+      setHistory({ previousPages });
+    },
+    openSearchIfNeeded: () => {
+      const currentPage = previousPages[previousPages.length - 1];
+      if (currentPage.isOnHomePage || currentPage.openedAssetShortHeader) {
+        previousPages.push({
+          isOnHomePage: false,
+          openedAssetShortHeader: null,
+          openedAssetPack: null,
+          filtersState: {
+            chosenCategory: null,
```

Comment:
can we factor this default state into a variable? something like `emptyFiltersState = { ... }` (because it's used multiple times)